### PR TITLE
feat: [CDS-37792]: Created dummy API for fetching users

### PR DESCRIPTION
**Pull request by mitishapandey**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35272)

### CODE REVIEW

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java [hunk 1]
```diff
@@ -52,6 +51,12 @@ public class JiraTaskNGParameters implements TaskParameters, ExecutionCapability
   String transitionToStatus; // required in a transition
   String transitionName; // optional - find a particular transition that goes to desired status
 
+  @Value
+  public class JiraUserSearchParams {
+    String accountId;
+    String userQuery;
+    String startAt;
+  }
   // Fields sent while creating/updating issue.
```
Comment (abhinav-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java:60]: this we have to change

**Review by arpit3107 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35272)* status: 12 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/35272#-N6w7pHz9lbBBqsyUSJ_:-N6w7pHz9lbBBqsyUSJa:b-2ejtd9) ([raw file](https://github.com/harness/harness-core/blob/4ee7f6caedb5e616989e906916db92113b7074b4/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java#L38)):*
> ```Java
>   JiraConnectorDTO jiraConnectorDTO;
>   List<EncryptedDataDetail> encryptionDetails;
>   JiraUserSearchParams jiraUserSearchParams;
> ```

is this change backward compatible? have you tested it with older version delegates?



_Code quote:_
```Java
JiraUserSearchParams jiraUserSearchParams;
```

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 5 of 9 files at r1, 2 of 2 files at r2, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35272)* status: :shipit: complete! ✓

**Review by arpit3107 — APPROVED** [review]
> Reviewed 5 of 9 files at r1, 2 of 2 files at r2, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35272)* status: :shipit: complete! ✓

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/35272#-N6w7pHz9lbBBqsyUSJ_:-N6wHYG61ll12wYGmqka:b-2uxucd) ([raw file](https://github.com/harness/harness-core/blob/4ee7f6caedb5e616989e906916db92113b7074b4/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java#L38)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

is this change backward compatible? have you tested it with older version delegates?
</blockquote></details>

This is behind FF. Approving.

**Comment by mitishapandey** [comment]
trigger codebasehash

**Comment by DeepakPatankar** [comment]
trigger codebasehash

File: 970-api-services-beans/src/main/java/io/harness/jira/JiraSearchUserMetadataNG.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+ */
+
+package io.harness.jira;
+
+public class JiraSearchUserMetadataNG {}
```
Comment (puthrayaharness) [inline 970-api-services-beans/src/main/java/io/harness/jira/JiraSearchUserMetadataNG.java:10]: Is this being used anywhere?

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35272)

**Pull request closed without merging**